File: lib/src/types/resolution_preset.dart
enum ResolutionPreset {
  low,
  medium,
  high,
  veryHigh,
  ultraHigh,
  max,
}

extension ResolutionPresetExtension on ResolutionPreset {
  String get name {
    switch (this) {
      case ResolutionPreset.low:
        return 'low';
      case ResolutionPreset.medium:
        return "medium";
      case ResolutionPreset.high:
        return "high";
      case ResolutionPreset.veryHigh:
        return "veryHigh";
      case ResolutionPreset.ultraHigh:
        return "ultraHigh";
      case ResolutionPreset.max:
        return "max";
      default:
        return null;
    }
  }
}

File: android/src/main/java/com/riguz/smart_camera/types/ResolutionPreset.java
package com.riguz.smart_camera.types;

import android.media.CamcorderProfile;

public enum ResolutionPreset {
    low(CamcorderProfile.QUALITY_QVGA),
    medium(CamcorderProfile.QUALITY_480P),
    high(CamcorderProfile.QUALITY_720P),
    veryHigh(CamcorderProfile.QUALITY_1080P),
    ultraHigh(CamcorderProfile.QUALITY_2160P),
    max(CamcorderProfile.QUALITY_HIGH);

    private final int quality;

    ResolutionPreset(int quality) {
        this.quality = quality;
    }

    public int getQuality() {
        return quality;
    }
}
